ERROR: failed packaging service 'backend': failed invoking event handlers for 'prepackage', 'prepackage' hook failed with exit code: '2', Path: 'Temp\azd-prepackage-3470486931.ps1'. : exit code: 2

<!--
IF SUFFICIENT INFORMATION IS NOT PROVIDED VIA THE FOLLOWING TEMPLATE THE ISSUE MIGHT BE CLOSED WITHOUT FURTHER CONSIDERATION OR INVESTIGATION
-->
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>

### Any log messages given by the failure
>

### Expected/desired behavior
>

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)

### azd version?
> run `azd version` and copy paste here.

### Versions
>

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

> frontend@0.0.0 build C:\Users\ChoubeTK\Downloads\workspace\app\frontend
> tsc && vite build

  |     ==| Packaging service backendsrc/authConfig.ts:3:64 - error TS2307: Cannot find module '@azure/msal-browser' or its corresponding type declarations.

3 import { AuthenticationResult, IPublicClientApplication } from "@azure/msal-browser";
                                                                 ~~~~~~~~~~~~~~~~~~~~~

src/authConfig.ts:77:13 - error TS7006: Parameter 'error' implicitly has an 'any' type.

77     .catch((error) => {
               ~~~~~

src/components/LoginButton/LoginButton.tsx:22:19 - error TS7006: Parameter 'error' implicitly has an 'any' type.

22           .catch((error) => console.log(error));
                     ~~~~~

src/components/LoginButton/LoginButton.tsx:30:17 - error TS7006: Parameter 'error' implicitly has an 'any' type.

30         .catch((error) => console.log(error));
                   ~~~~~

src/index.tsx:6:65 - error TS2307: Cannot find module '@azure/msal-browser' or its corresponding type declarations.

6 import { PublicClientApplication, EventType, AccountInfo } from '@azure/msal-browser';
                                                                  ~~~~~~~~~~~~~~~~~~~~~

src/index.tsx:25:36 - error TS7006: Parameter 'event' implicitly has an 'any' type.

25     msalInstance.addEventCallback((event) => {
                                      ~~~~~


Found 6 errors in 3 files.

Errors  Files
     2  src/authConfig.ts:3
     2  src/components/LoginButton/LoginButton.tsx:22
     2  src/index.tsx:6
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! frontend@0.0.0 build: `tsc && vite build`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the frontend@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ChoubeTK\AppData\Roaming\npm-cache\_logs\2023-09-27T00_24_36_002Z-debug.log
  (x) Failed: Packaging service backend

ERROR: failed packaging service 'backend': failed invoking event handlers for 'prepackage', 'prepackage' hook failed with exit code: '2', Path: 'C:\Users\ChoubeTK\AppData\Local\Temp\azd-prepackage-4273095110.ps1'. : exit code: 2

I have the same error when running azd up. Here's my log:

 Try `npm i --save-dev @types/fluentui__react-table` if it exists or add a new declaration (.d.ts) file containing `declare module '@fluentui/react-table';`        

12 } from '@fluentui/react-table';
          ~~~~~~~~~~~~~~~~~~~~~~~

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:46:25 - error TS7006: Parameter 'item' implicitly has an 'any' type.

46             renderCell: item => {
                           ~~~~

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:58:25 - error TS7006: Parameter 'item' implicitly has an 'any' type.

58             renderCell: item => {
                           ~~~~

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:71:27 - error TS7006: Parameter 'item' implicitly has an 'any' type.

71                 getRowId={item => item.name}
                             ~~~~

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:75:25 - error TS7031: Binding element 'renderHeaderCell' implicitly has an 'any' type.

75                     {({ renderHeaderCell }) => <DataGridHeaderCell>{renderHeaderCell()}</DataGridHeaderCell>}
                           ~~~~~~~~~~~~~~~~

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:79:21 - error TS7031: Binding element 'item' implicitly has an 'any' type.

79                 {({ item, rowId }) => (
                       ~~~~

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:79:27 - error TS7031: Binding element 'rowId' implicitly has an 'any' type.

79                 {({ item, rowId }) => (
                             ~~~~~

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:81:25 - error TS7031: Binding element 'renderCell' implicitly has an 'any' type.

81                     {({ renderCell }) => <DataGridCell>{renderCell(item)}</DataGridCell>}
                           ~~~~~~~~~~


Found 8 errors in the same file, starting at: src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:12

  (x) Failed: Packaging service backend

ERROR: failed packaging service 'backend': failed invoking event handlers for 'prepackage', 'prepackage' hook failed with exit code: '2', Path: 'C:\Users\XXXX\XXX\Local\Temp\azd-prepackage-2967539339.ps1'. : exit code: 2

Thank you Pamela, I tried but still getting this error:

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:71:61 - error TS7031: Binding element 'renderCell' implicitly has an 'any' type.

71                         <DataGridRow<Claim> key={rowId}>{({ renderCell }) => <DataGridCell>{renderCell(item)}</DataGridCell>}</DataGridRow>
                                                               ~~~~~~~~~~


Found 2 errors in the same file, starting at: src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:13

  (x) Failed: Packaging service backend

ERROR: failed packaging service 'backend': failed invoking event handlers for 'prepackage', 'prepackage' hook failed with exit code: '2', Path: 'C:\Users\XXXX\XXX\Local\Temp\azd-prepackage-189254659.ps1'. : exit code: 2

Updates, I solved the two by doing "npm i @fluentui/react-table" and claiming the item also for the renderCell in line 72:
 {({ renderCell }: { renderCell: (item: Claim) => React.ReactNode }) => <DataGridCell>{renderCell(item)}</DataGridCell>}.

However, now I am getting this: 
 error TS2786: 'DataGrid' cannot be used as a JSX component.
  Its type 'ForwardRefExoticComponent<DataGridProps & RefAttributes<HTMLDivElement>>' is not a valid JSX element type.
    Type 'ForwardRefExoticComponent<DataGridProps & RefAttributes<HTMLDivElement>>' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Type 'import("C:/Users/XXX/XXX/XX/XX/node_modules/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'.
        Type 'ReactElement<any, string | JSXElementConstructor<any>>' is not assignable to type 'ReactNode'.
          Property 'children' is missing in type 'ReactElement<any, string | JSXElementConstructor<any>>' but required in type 'ReactPortal'.

63             <DataGrid items={items} columns={columns} sortable getRowId={(item: Claim) => item.name}>
                ~~~~~~~~

  node_modules/@types/react/index.d.ts:198:9
    198         children: ReactNode;
                ~~~~~~~~
    'children' is declared here.

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:64:18 - error TS2786: 'DataGridHeader' cannot be used as a JSX component.
  Its type 'ForwardRefExoticComponent<TableHeaderProps & RefAttributes<HTMLDivElement>>' is not a valid JSX element type.
    Type 'ForwardRefExoticComponent<TableHeaderProps & RefAttributes<HTMLDivElement>>' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Type 'import("XXX/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'.

64                 <DataGridHeader>
                    ~~~~~~~~~~~~~~

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:66:25 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Omit<TableRowProps, "children"> & Omit<ComponentProps<DataGridRowSlots>, "children"> & { children: CellRenderFunction<unknown>; } & RefAttributes<...>): ReactNode', gave the following error.
    Type '({ renderHeaderCell }: { renderHeaderCell: () => React.ReactNode; }) => JSX.Element' is not assignable to type 'CellRenderFunction<unknown>'.
      Types of parameters '__0' and 'column' are incompatible.
        Type 'TableColumnDefinition<unknown>' is not assignable to type '{ renderHeaderCell: () => ReactNode; }'.
          The types returned by 'renderHeaderCell()' are incompatible between these types.
            Type 'import("XXX/node_modules/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'.
  Overload 2 of 2, '(props: DataGridRowProps<unknown>): Element', gave the following error.
    Type '({ renderHeaderCell }: { renderHeaderCell: () => React.ReactNode; }) => JSX.Element' is not assignable to type 'CellRenderFunction<unknown>'.

66                         {({ renderHeaderCell }: { renderHeaderCell: () => React.ReactNode }) => <DataGridHeaderCell>{renderHeaderCell()}</DataGridHeaderCell>}
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  ../../node_modules/@fluentui/react-table/dist/index.d.ts:242:5
    242     children: CellRenderFunction<TItem>;
            ~~~~~~~~
    The expected type comes from property 'children' which is declared here on type 'IntrinsicAttributes & Omit<TableRowProps, "children"> & Omit<ComponentProps<DataGridRowSlots>, "children"> & { ...; } & RefAttributes<...>'
  ../../node_modules/@fluentui/react-table/dist/index.d.ts:242:5
    242     children: CellRenderFunction<TItem>;
            ~~~~~~~~
    The expected type comes from property 'children' which is declared here on type 'IntrinsicAttributes & Omit<TableRowProps, "children"> & Omit<ComponentProps<DataGridRowSlots>, "children"> & { ...; }'

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:66:98 - error TS2786: 'DataGridHeaderCell' cannot be used as a JSX component.
  Its type 'ForwardRefExoticComponent<DataGridHeaderCellProps & RefAttributes<HTMLDivElement>>' is not a valid JSX element type.
    Type 'ForwardRefExoticComponent<DataGridHeaderCellProps & RefAttributes<HTMLDivElement>>' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Type 'import("XXX/node_modules/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'.

66                         {({ renderHeaderCell }: { renderHeaderCell: () => React.ReactNode }) => <DataGridHeaderCell>{renderHeaderCell()}</DataGridHeaderCell>}
                                                                                                    ~~~~~~~~~~~~~~~~~~

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:72:29 - error TS2322: Type '({ renderCell }: { renderCell: (item: Claim) => React.ReactNode; }) => JSX.Element' is not assignable to type 'CellRenderFunction<Claim>'.
  Types of parameters '__0' and 'column' are incompatible.
    Type 'TableColumnDefinition<Claim>' is not assignable to type '{ renderCell: (item: Claim) => ReactNode; }'.
      The types returned by 'renderCell(...)' are incompatible between these types.
        Type 'import("XXX/node_modules/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'.

72                             {({ renderCell }: { renderCell: (item: Claim) => React.ReactNode }) => <DataGridCell>{renderCell(item)}</DataGridCell>}
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  ../../node_modules/@fluentui/react-table/dist/index.d.ts:242:5
    242     children: CellRenderFunction<TItem>;
            ~~~~~~~~
    The expected type comes from property 'children' which is declared here on type 'IntrinsicAttributes & Omit<TableRowProps, "children"> & Omit<ComponentProps<DataGridRowSlots>, "children"> & { ...; }'

src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:72:101 - error TS2786: 'DataGridCell' cannot be used as a JSX component.
  Its type 'ForwardRefExoticComponent<DataGridCellProps & RefAttributes<HTMLDivElement>>' is not a valid JSX element type.
    Type 'ForwardRefExoticComponent<DataGridCellProps & RefAttributes<HTMLDivElement>>' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Type 'import("XXX/node_modules/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'.

72                             {({ renderCell }: { renderCell: (item: Claim) => React.ReactNode }) => <DataGridCell>{renderCell(item)}</DataGridCell>}
                                                                                                       ~~~~~~~~~~~~


Found 6 errors in the same file, starting at: src/components/TokenClaimsDisplay/TokenClaimsDisplay.tsx:63

  (x) Failed: Packaging service backend

ERROR: failed packaging service 'backend': failed invoking event handlers for 'prepackage', 'prepackage' hook failed with exit code: '2', Path: 'XXX\AppData\Local\Temp\azd-prepackage-477418279.ps1'. : exit code: 2

--- a/app/frontend/package.json
+++ b/app/frontend/package.json
@@ -12,7 +12,8 @@
     "watch": "tsc && vite build --watch"
   },
   "dependencies": {
-    "@azure/msal-react": "^2.0.2",
+    "@azure/msal-react": "^2.0.4",
+    "@azure/msal-browser": "^3.1.0",
     "@fluentui/react": "^8.110.7",
     "@fluentui/react-components": "^9.33.0",
     "@fluentui/react-icons": "^2.0.206",

